src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBResource.cs

```diff
@@ -74,22 +74,36 @@ void IResourceWithAzureFunctionsConfig.ApplyAzureFunctionsConfiguration(IDiction
     {
         if (IsEmulator || UseAccessKeyAuthentication)
         {
-            // Injected to support Azure Functions listener initialization.
-            target[connectionName] = ConnectionStringExpression;
-            // Injected to support Aspire client integration for CosmosDB in Azure Functions projects.
-            target[$"Aspire__Microsoft__Azure__Cosmos__{connectionName}__ConnectionString"] = ConnectionStringExpression;
-            target[$"Aspire__Microsoft__EntityFrameworkCore__Cosmos__{connectionName}__ConnectionString"] = ConnectionStringExpression;
+            SetConnectionString(target, connectionName, ConnectionStringExpression);
         }
         else
         {
-            // Injected to support Azure Functions listener initialization.
-            target[$"{connectionName}__accountEndpoint"] = ConnectionStringExpression;
-            // Injected to support Aspire client integration for CosmosDB in Azure Functions projects.
-            target[$"Aspire__Microsoft__Azure__Cosmos__{connectionName}__AccountEndpoint"] = ConnectionStringExpression;
-            target[$"Aspire__Microsoft__EntityFrameworkCore__Cosmos__{connectionName}__AccountEndpoint"] = ConnectionStringExpression;
+            SetAccountEndpoint(target, connectionName);
         }
     }
 
+    internal void SetConnectionString(IDictionary<string, object> target, string connectionName, ReferenceExpression connectionStringExpression)
+    {
+        // Always inject the connection string associated with the top-level resource
+        // for the Azure Functions host.
+        target[connectionName] = ConnectionStringExpression;
+        // Injected to support Aspire client integration for CosmosDB in Azure Functions projects.
+        // Use the child resource connection string here to support child resource integrations.
+        target[$"Aspire__Microsoft__EntityFrameworkCore__Cosmos__{connectionName}__ConnectionString"] = connectionStringExpression;
+        target[$"Aspire__Microsoft__Azure__Cosmos__{connectionName}__ConnectionString"] = connectionStringExpression;
+    }
+
+    internal void SetAccountEndpoint(IDictionary<string, object> target, string connectionName)
+    {
+        // Always inject the connection string associated with the top-level resource
+        // for the Azure Functions host.
+        target[$"{connectionName}__accountEndpoint"] = ConnectionStringExpression;
+        // Injected to support Aspire client integration for CosmosDB in Azure Functions projects.
+        // Use the child resource connection string here to support child resource integrations.
```

Comment:
What does this comment mean? `Use the child resource connection string here`, but the code uses `this.ConnectionStringExpression`.

Reply:
Yes, this is copy-pasta. 🍝